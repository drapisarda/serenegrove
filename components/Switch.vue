<template>
  <div class="toggle-switch">
    <label class="toggle-label" :for="id">
      {{ label }}
      <input :id="id" v-model="model" type="checkbox" class="toggle-input" />
    </label>
  </div>
</template>

<script lang="ts" setup>
import { computed, defineEmits } from 'vue'
const props = defineProps({
  modelValue: {
    type: Boolean,
    default: false,
  },
  label: {
    type: String,
    default: '',
  },
  id: {
    type: String,
    default: '',
  },
})

const emit = defineEmits(['update:modelValue'])
const model = computed({
  get() {
    return props.modelValue
  },
  set(value) {
    emit('update:modelValue', value, props.modelValue)
  },
})
</script>

<style lang="scss">
// https://uiverse.io/ClawHack1/itchy-bobcat-18 edited to be fully HTM compliant

.toggle-switch {
  position: relative;
  display: inline-block;
  width: 40px;
  height: 24px;
  margin: 10px;

  .toggle-input {
    display: none;
  }

  .toggle-label {
    position: absolute;
    top: 0;
    left: 0;
    width: 40px;
    height: 24px;
    background-color: #ccc;
    border-radius: 34px;
    cursor: pointer;
    transition: background-color 0.3s;

    &::before {
      content: '';
      position: absolute;
      width: 20px;
      height: 20px;
      border-radius: 50%;
      top: 2px;
      left: 2px;
      background-color: #fff;
      box-shadow: 0px 2px 5px 0px rgba(0, 0, 0, 0.3);
      transition: transform 0.3s;
    }
  }

  .toggle-label:has(.toggle-input:checked) {
    background-color: #4caf50;

    &::before {
      transform: translateX(16px);
    }
  }

  &.light {
    .toggle-label {
      background-color: #bebebe;
    }

    .toggle-label:has(.toggle-input:checked) {
      background-color: #9b9b9b;

      &::before {
        transform: translateX(6px);
      }
    }
  }

  &.dark {
    .toggle-label {
      background-color: #4b4b4b;
    }

    .toggle-label:has(.toggle-input:checked) {
      background-color: #717171;

      &::before {
        transform: translateX(16px);
      }
    }
  }
}
</style>
