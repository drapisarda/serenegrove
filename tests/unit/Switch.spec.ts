import Switch from '@/components/Switch.vue'
import { mount, type VueWrapper } from '@vue/test-utils'
import { expect } from 'vitest'

describe('Switch component test', async () => {
  let wrapper: VueWrapper

  beforeAll(() => {
    wrapper = mount(Switch, {
      props: {
        modelValue: false,
        label: 'Lorem Label',
        id: 'loremId',
      },
    })
  })

  test('Switch mounts', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })

  test('Switch changes value and emits properly', async () => {
    expect(wrapper.emitted()['update:modelValue']).toBe(undefined)

    const firstValue = true
    const secondValue = false
    await wrapper.get('input').setValue(firstValue)
    await wrapper.get('input').setValue(secondValue)

    expect(wrapper.emitted()['update:modelValue'][0][0]).toBe(firstValue)
    expect(wrapper.emitted()['update:modelValue'][0][1]).toBe(secondValue)
  })
})
