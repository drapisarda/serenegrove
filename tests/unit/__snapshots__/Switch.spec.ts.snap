// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch component test > Switch mounts 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="toggle-switch"
      >
        <label
          class="toggle-label"
          for="loremId"
        >
          Lorem Label 
          <input
            class="toggle-input"
            id="loremId"
            type="checkbox"
          />
        </label>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "slotsProxy": null,
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "id": "loremId",
            "label": "Lorem Label",
            "modelValue": false,
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "emit": [Function],
            "model": ComputedRefImpl {
              "__v_isReadonly": false,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": false,
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    Set {
                      [Circular],
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
            "props": {
              "id": "loremId",
              "label": "Lorem Label",
              "modelValue": false,
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": {
            "update:modelValue": null,
          },
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "id": "loremId",
            "label": "Lorem Label",
            "modelValue": false,
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "id": {
                "0": false,
                "1": true,
                "default": "",
                "type": [Function],
              },
              "label": {
                "0": false,
                "1": true,
                "default": "",
                "type": [Function],
              },
              "modelValue": {
                "0": true,
                "1": true,
                "default": false,
                "type": [Function],
              },
            },
            [
              "modelValue",
              "label",
              "id",
            ],
          ],
          "provides": {},
          "proxy": {
            "id": "loremId",
            "label": "Lorem Label",
            "modelValue": false,
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": false,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": false,
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        Set {
                          [Circular],
                        },
                        Set {
                          [Circular],
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "emit": [Function],
            "model": false,
            "props": {
              "id": "loremId",
              "label": "Lorem Label",
              "modelValue": false,
            },
          },
          "slots": {},
          "slotsProxy": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Lorem Label ",
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Lorem Label ,
                    "key": null,
                    "patchFlag": 1,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(v-txt),
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": [
                      {
                        "arg": undefined,
                        "dir": {
                          "beforeUpdate": [Function],
                          "created": [Function],
                          "deep": true,
                          "getSSRProps": [Function],
                          "mounted": [Function],
                        },
                        "instance": {},
                        "modifiers": {},
                        "oldValue": undefined,
                        "value": false,
                      },
                    ],
                    "dynamicChildren": null,
                    "dynamicProps": [
                      "id",
                    ],
                    "el": <input
                      class="toggle-input"
                      id="loremId"
                      type="checkbox"
                    />,
                    "key": null,
                    "patchFlag": 8,
                    "props": {
                      "class": "toggle-input",
                      "id": "loremId",
                      "onUpdate:modelValue": [Function],
                      "type": "checkbox",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "input",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "for",
                ],
                "el": <label
                  class="toggle-label"
                  for="loremId"
                >
                  Lorem Label 
                  <input
                    class="toggle-input"
                    id="loremId"
                    type="checkbox"
                  />
                </label>,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "class": "toggle-label",
                  "for": "loremId",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Lorem Label ",
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": Lorem Label ,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(v-txt),
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "deep": true,
                      "getSSRProps": [Function],
                      "mounted": [Function],
                    },
                    "instance": {},
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": false,
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": [
                  "id",
                ],
                "el": <input
                  class="toggle-input"
                  id="loremId"
                  type="checkbox"
                />,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "class": "toggle-input",
                  "id": "loremId",
                  "onUpdate:modelValue": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Lorem Label ",
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Lorem Label ,
                    "key": null,
                    "patchFlag": 1,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(v-txt),
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": [
                      {
                        "arg": undefined,
                        "dir": {
                          "beforeUpdate": [Function],
                          "created": [Function],
                          "deep": true,
                          "getSSRProps": [Function],
                          "mounted": [Function],
                        },
                        "instance": {},
                        "modifiers": {},
                        "oldValue": undefined,
                        "value": false,
                      },
                    ],
                    "dynamicChildren": null,
                    "dynamicProps": [
                      "id",
                    ],
                    "el": <input
                      class="toggle-input"
                      id="loremId"
                      type="checkbox"
                    />,
                    "key": null,
                    "patchFlag": 8,
                    "props": {
                      "class": "toggle-input",
                      "id": "loremId",
                      "onUpdate:modelValue": [Function],
                      "type": "checkbox",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "input",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "for",
                ],
                "el": <label
                  class="toggle-label"
                  for="loremId"
                >
                  Lorem Label 
                  <input
                    class="toggle-input"
                    id="loremId"
                    type="checkbox"
                  />
                </label>,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "class": "toggle-label",
                  "for": "loremId",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="toggle-switch"
            >
              <label
                class="toggle-label"
                for="loremId"
              >
                Lorem Label 
                <input
                  class="toggle-input"
                  id="loremId"
                  type="checkbox"
                />
              </label>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "toggle-switch",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/danielerapisarda/projects/serenegrove/components/Switch.vue",
            "__name": "Switch",
            "components": {},
            "emits": [
              "update:modelValue",
            ],
            "props": {
              "id": {
                "default": "",
                "type": [Function],
              },
              "label": {
                "default": "",
                "type": [Function],
              },
              "modelValue": {
                "default": false,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="toggle-switch"
        >
          <label
            class="toggle-label"
            for="loremId"
          >
            Lorem Label 
            <input
              class="toggle-input"
              id="loremId"
              type="checkbox"
            />
          </label>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "id": "loremId",
          "label": "Lorem Label",
          "modelValue": false,
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/danielerapisarda/projects/serenegrove/components/Switch.vue",
          "__name": "Switch",
          "components": {},
          "emits": [
            "update:modelValue",
          ],
          "props": {
            "id": {
              "default": "",
              "type": [Function],
            },
            "label": {
              "default": "",
              "type": [Function],
            },
            "modelValue": {
              "default": false,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="toggle-switch"
        >
          <label
            class="toggle-label"
            for="loremId"
          >
            Lorem Label 
            <input
              class="toggle-input"
              id="loremId"
              type="checkbox"
            />
          </label>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.11",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <div
    class="toggle-switch"
  >
    <label
      class="toggle-label"
      for="loremId"
    >
      Lorem Label 
      <input
        class="toggle-input"
        id="loremId"
        type="checkbox"
      />
    </label>
  </div>,
}
`;
